gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -37,19 +38,23 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.Nullable;
 
 /**
  * A {@link ManagedChannel} that will send requests round robin via a set of channels.
  *
  * <p>Package-private for internal use.
  */
 class ChannelPool extends ManagedChannel {
+  private static final int EXECUTOR_THREAD_POOL_SIZE = 2;
   private final ImmutableList<ManagedChannel> channels;
   private final AtomicInteger indexTicker = new AtomicInteger();
   private final String authority;
+  @Nullable private ScheduledExecutorService executorService;
```

Comment:
Here and everywhere else, please name it more specifically to avoid confusion

Reply:
Please make it clear that this this class will end up owning the executor and shut it down on pool shutdown